### DISCUSSION TITLE
How to manualy trigger ra cargo check via hotkey

### DISCUSSION THREAD

**Discussion by KappaShilaff** [zed-industries/zed#18925] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I can't find how to do a cargo check on save, but triggered by a hotkey. This is critically important for a laptop to avoid consuming too much battery on rechecks for every save. A regular spawn task is not what I need because it doesn't highlight the syntax in the code itself.

for example code like
let int: u32 = "strrrrrr";

won't be underlined with red line unless cargo check was run. Maybe it's possible to add some debounce so cargo check won't be running every second?

**Comment by SomeoneToIgnore** [comment]
Zed has tasks, which allow running relatively arbitrary scripts inside the project/other directory: you could create your own or use the ones provided by rust integration.

https://zed.dev/docs/tasks#custom-keybindings-for-tasks shows how to add a keybindings on top 

-------------

"cargo check on every second" are the defaults of rust-analyzer (see `rust-analyzer.checkOnSave (default: true)` in https://rust-analyzer.github.io/manual.html)
The reason for this is the fact that there's not that many "stable" diagnostics, implemented by rust-analyzer itself and most of them are coming from `cargo check` output parsed inside rust-analyzer first.

https://zed.dev/docs/languages/rust#large-projects-and-performance mentions how to disable cargo check entirely, so you can use tasks (and cmd+click in the terminal) to fix errors continuously.

**Reply by KappaShilaff** [reply]
> task is not what I need because it doesn't highlight the syntax in the code itself.

Yes, you can trigger tasks via custom bindings, but it won't give you diagnostics in the IDE. I want to trigger cargo check with its output parsed by RA, so it will show in red what's wrong in my code. Alt + clicking is not the best experience.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
